pkg/cmd/proxy.go

```diff
@@ -0,0 +1,143 @@
+/*
+Copyright 2020 The KubeOne Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+
+	"k8c.io/kubeone/pkg/state"
+)
+
+type proxyOpts struct {
+	globalOptions
+	ListenAddr string `longflag:"listen"`
+}
+
+func proxyCmd(rootFlags *pflag.FlagSet) *cobra.Command {
+	opts := &proxyOpts{}
+
+	cmd := &cobra.Command{
+		Use:   "proxy",
+		Short: "Proxy to the kube-apiserver using SSH tunnel",
+		Long: `
+HTTPS Proxy (CONNECT method) SSH tunnel.
+`,
```

Comment:
Should we include some details when this command is used, i.e. mention that you need this if your load balancer is not publicly accessible?

Reply:
done